cpp/misra/src/rules/RULE-21-10-2/NoCsetjmpHeader.ql

```diff
@@ -0,0 +1,53 @@
+/**
+ * @id cpp/misra/no-csetjmp-header
+ * @name RULE-21-10-2: The standard header file <csetjmp> shall not be used
+ * @description Using facilities from the <csetjmp> header causes undefined behavior by bypassing
+ *              normal function return mechanisms and may result in non-trivial object destruction
+ *              being omitted.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/misra/id/rule-21-10-2
+ *       scope/single-translation-unit
+ *       external/misra/enforcement/decidable
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.cpp.misra
+import codingstandards.cpp.BannedFunctions
+
+class CSetJmpHeader extends Include {
+  CSetJmpHeader() { this.getIncludeText().regexpMatch("[<\\\"](csetjmp|setjmp.h)[>\\\"]") }
+}
+
+class JmpBufVariable extends Variable {
+  JmpBufVariable() { this.getType().(UserType).hasGlobalOrStdName("jmp_buf") }
```

Comment:
Another example of missing type uses, though in theory inconsequential if we detect calls to setjmp / longjmp